### ISSUE TITLE
Key management tab broken when GNUPG mode enabled

### ISSUE THREAD

**Issue by nezetic** [mailvelope/mailvelope#638]
OS: macOS 10.14.2
Browser: Chrome 71.0.3578.98
Mailvelope: 3.0.2
GPG: 
```
OK GPGME-Tool 1.12.0 ready
ENGINE
S ENGINE OpenPGP /usr/local/MacGPG2/bin/gpg 2.2.10 1.4.0
S ENGINE CMS /usr/local/MacGPG2/bin/gpgsm 2.2.10 2.0.4
S ENGINE GPGCONF /usr/local/MacGPG2/bin/gpgconf 2.2.10 2.0.4
S ENGINE Assuan ~/.gnupg/S.gpg-agent 1.0.0 1.0.0 !GPG_AGENT
S ENGINE UIServer ~/.gnupg/S.uiserver 1.0.0 1.0.0
S ENGINE Spawn /nonexistent 1.0.0 1.0.0
```

Enabling GNUPG integration is breaking the _Key Management_ tab. Even switching back to _OpenGPG.js_ doesn't solve this error (only re-installing the extension does).

```
Uncaught (in promise) {message: "Cannot read property 'getAttr' of undefined", code: "INTERNAL_ERROR"}
Promise.then (async)
  | loadKeyring @ app.bundle.js:34398
  | componentDidMount @ app.bundle.js:34382
  | async function (async)
  | componentDidMount | @ | app.bundle.js:34381
```

**Comment by nezetic** [comment]
Ok, I found the issue. I've compiled _gpgme-json_ in order to work with _GPGTools_ (as this is what I'm using). But the path is not standard, and as a consequence, without proper environment variables, _gpgme-json_ will not work properly.

```
# echo '{"op":"version"}' | env -i /usr/local/MacGPG2/bin/gpgme-json -i
gpgme-json 1.12.0 ready (enter ",help" for help)
> >
===> {
===> 	"gpgme":	"1.12.0",
===> 	"info":	[{
===> 			"protocol":	"spawn",
===> 			"fname":	"/nonexistent",
===> 			"version":	"1.0.0",
===> 			"req_version":	"1.0.0",
===> 			"homedir":	"default"
===> 		}]
===> }
```

If I'm using this wrapper in the Native Messaging host, everything is working as expected.

/usr/local/MacGPG2/bin/gpgme-json.sh
```bash
#!/bin/sh
export PATH=$PATH:/usr/local/MacGPG2/bin
/usr/local/MacGPG2/bin/gpgme-json $@
```

**Comment by toberndo** [comment]
Thanks for reporting this issue. We need to improve failure resistance, a broken gpgme-json should not break the Mailvelope UI.

The path issue I think could be solved with the `--enable-fixed-path=/usr/local/MacGPG2/bin`option when you compile GPGME.

**Comment by nezetic** [comment]
Indeed, compiling with this option solve the issue (and is much cleaner).

**Comment by doubleo2** [comment]
@nezetic How did you install gpgme to use GPGTools?

When installing gpgme from source, I tried running `./configure --enable-fixed-path=/usr/local/MacGPG2/bin`, but it complains that _libgpg-error_ and _libassuan_ are missing. Adding `--with-lib{gpg-error,assuan}-prefix=/usr/local/MacGPG2` doesn't seem to work either because it is looking for the _gpg-error-config_ and _libassuan-config_ scripts in the bin folder and GPG Suite doesn't have them.

**Comment by nezetic** [comment]
@doubleo2 I don't use the GPGTools binary release (which probably doesn't provide headers & build scripts), as I prefer to compile GPG myself.

Thanks a lot @toberndo for improving the keyring handling.

**Comment by doubleo2** [comment]
@nezetic Ah, that explains it. 

For anyone else who finds this issue, I was able to get it working by compiling the two missing dependencies (libgpg-error and libassuan) from source before compiling gpgme with the `--enabled-fixed-path` option. Everything seems to be working now.
